base/src/com/google/idea/blaze/base/sync/ProjectStateSyncTask.java

```diff
@@ -125,7 +126,7 @@ private SyncProjectState getProjectState(BlazeContext context, BlazeSyncParams p
         createBazelInfoFuture(context, syncFlags, params.syncMode());
 
     ListenableFuture<WorkingSet> workingSetFuture;
-    if (params.addWorkingSet() || params.syncMode() == SyncMode.FULL) {
+    if (params.addWorkingSet() || (BlazeUserSettings.getInstance().getExpandSyncToWorkingSet() && params.syncMode() == SyncMode.FULL)) {
```

Comment:
Don't know when this one is used but it is bad to ignore the setting anyway so we need to double check here.

Reply:
Oh, params.addWorkingSet already checks for that, but I'll merge

Reply:
I want to avoid unconditional `params.syncMode() == SyncMode.FULL` this is quite recent change from September